src/private/CABasicAnimation+MotionAnimator.m

```diff
@@ -20,6 +20,22 @@
 
 #import <UIKit/UIKit.h>
 
+#pragma mark - Private
+
+static BOOL IsNumberValue(id someValue) {
+  return [someValue isKindOfClass:[NSNumber class]];
+}
+
+static BOOL IsValueType(id someValue, char *objCType) {
+  if ([someValue isKindOfClass:[NSValue class]]) {
+    NSValue *asValue = (NSValue *)someValue;
+    return strcmp(asValue.objCType, objCType) == 0;
```

Comment:
Please switch to strncmp and limit the length to something short like 10 characters.
CGSize = "{CGSize=dd}"
CGRect = "{CGRect={CGPoint=dd}{CGSize=dd}}"
CGPoint = "{CGPoint=dd}"
CGRectNull = "{CGRect={CGPoint=dd}{CGSize=dd}}"

Reply:
Fixed by getting objCType's string length. objCType is "trusted" in that we're only ever providing `@encode` values to it.